src/Storage/Authorization/AuthorizationService.cs

```diff
@@ -152,6 +149,48 @@ public async Task<bool> AuthorizeInstanceAction(Instance instance, string action
             return authorized;
         }
 
+        /// <inheritdoc/>>
+        public async Task<bool> AuthorizeAnyOfInstanceActions(Instance instance, List<string> actions, string task = null)
+        {
+            string org = instance.Org;
+            string app = instance.AppId.Split('/')[1];
+            int instanceOwnerPartyId = int.Parse(instance.InstanceOwner.PartyId);
+            XacmlJsonRequestRoot request;
+            if (actions.Count == 0)
+            {
+                return false;
+            }
+
+            ClaimsPrincipal user = _claimsPrincipalProvider.GetUser();
+            if (instance.Id == null)
+            {
+                request = DecisionHelper.CreateDecisionRequest(org, app, user, actions[0], instanceOwnerPartyId, null);
+            }
+            else
+            {
+                Guid instanceGuid = Guid.Parse(instance.Id.Split('/')[1]);
+                request = DecisionHelper.CreateDecisionRequest(org, app, user, actions[0], instanceOwnerPartyId, instanceGuid, task);
+            }
+
+            for (int i = 1; i < actions.Count; i++)
+            {
+                request.Request.Action.Add(DecisionHelper.CreateActionCategory(actions[i]));
```

Comment:
This will probably not work. We would need to create individual DecisionRequests for each action. We have examples of this in relation to MessageBox functionality. We might be able to use the CreateMultiDecisionRequest method almost as is.